### PR TITLE
go: update completions for 1.18

### PR DESCRIPTION
**Pull request by seankhliao**
update completions for `go`, adding new subcommands (`version`, `work *`) and flags

### CODE REVIEW

File: src/_golang [hunk 1]
```diff
@@ -393,15 +407,22 @@ case $state in
         _arguments \
           '-i[remove corresponding installed archive or binary]' \
           '-r[apply clean recursively on all dependencies]' \
+          '-cache[remove the entire go build cache]' \
+          '-testcache[expire all test results in thego build cache]' \
+          '-modcache[clean to remove the entire moduledownload cach]' \
```
Comment (syohex) [inline src/_golang:412]:
```suggestion
          '-testcache[expire all test results in the go build cache]' \
          '-modcache[clean to remove the entire module download cache]' \
```

**Review by syohex — COMMENTED** [review]
Thanks. I think this PR is almost good. I confirmed this works.
Could you check some typo that I commented ?

**Comment by seankhliao** [comment]
thanks for the quick review, I committed the typo fixes

**Comment by syohex** [comment]
Thanks

**Pull request merged**